File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        String[] words1 = s1.split(" ");
        String[] words2 = s2.split(" ");
        Map<String, Integer> countMap = new HashMap<>();

        for (String word : words1) {
            countMap.put(word, countMap.getOrDefault(word, 0) + 1);
        }

        for (String word : words2) {
            countMap.put(word, countMap.getOrDefault(word, 0) + 1);
        }

        List<String> uncommonWords = new ArrayList<>();
        for (Map.Entry<String, Integer> entry : countMap.entrySet()) {
            if (entry.getValue() == 1) {
                uncommonWords.add(entry.getKey());
            }
        }

        return uncommonWords.toArray(new String[0]);
    }
    }

File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> list[str]:
        dic={}
        for i in list(s1.split()):
            dic[i]=dic.get(i,0)+1
        for j in list(s2.split()):
            dic[j]=dic.get(j,0)+1
        return [i for i in dic if dic[i]==1]
            

        